lib/topological_inventory/orchestrator/worker.rb

```diff
@@ -1,124 +1,168 @@
-            @object_manager.delete_secret("#{deployment.metadata.name}-secrets")
+
+
+      ### API STUFF
+      def each_endpoint
+        return enum_for(:each_endpoint) unless block_given?
+        each_resource(url_for("source_types")) do |source_type|
+          collector_definition = collector_definitions[source_type["name"]]
+          next if collector_definition.nil?
+          each_resource(url_for("source_types/#{source_type["id"]}/sources")) do |source|
+            each_resource(url_for("sources/#{source["id"]}/endpoints")) do |endpoint|
+              each_resource(url_for("authentications?resource_type=Endpoint&resource_id=#{endpoint["id"]}&authtype=default")) do |authentication|
+                definition_information = collector_definition[endpoint["role"]]
+                next unless definition_information
+                yield source, endpoint, authentication, definition_information
+              end
+            end
           end
         end
       end
 
-      def have_deployment_for_source?(source)
-        collector_deployments.detect do |deployment|
-          source_for_deployment(deployment) == source["uid"]
+      def collectors_from_database(hash)
+        each_endpoint.collect do |source, endpoint, authentication, definition_information|
+          auth = authentication_with_password(authentication["id"])
+          value = {
+              "host"       => endpoint["host"],
+              "image"      => definition_information["image"],
+              "source_id"  => source["id"],
+              "source_uid" => source["uid"],
+              "secret"     => {
+                "password" => auth["password"],
+                "username" => auth["username"],
+              },
+            }
+          key = digest(value)
+          hash[key] = value
+          key
         end
       end
 
-      def create_objects_for_source(source)
-        @object_manager.create_deployment_config(deployment_name_for_source(source)) do |d|
-          container = d[:spec][:template][:spec][:containers].first
-          container[:env] << collector_container_environment(source)
-          container[:image] = "#{ENV["MY_NAMESPACE"]}/topological-inventory-collector-openshift:latest"
-        end
+      def url_for(path)
+        File.join(@api_base_url, path)
       end
 
-      private
-
-      def collector_container_environment(source)
+      def each_resource(url, &block)
+        return if url.nil?
+        response = JSON.parse(RestClient.get(url))
+        response["data"].each { |i| yield i }
+        each_resource(response["links"]["next"], &block)
       end
 
-      def secret_name_for_source(source)
-        "#{deployment_name_for_source(source)}-secrets"
+
+      # HACK for Authentications
+      def authentication_with_password(id)
+        url = URI.parse(@api_base_url).tap do |uri|
+          uri.path = "/internal/v0.0/authentications/#{id}"
```

Comment:
Is this the only version for the internal API?

Reply:
yes